### DISCUSSION TITLE
any one try curl with cmake build?

### DISCUSSION THREAD

**Discussion by calvin2021y** [curl/curl#11156] [Q&A]
```sh
CMake Error at CMakeLists.txt:1221 (if):
  if given arguments:

    "GREATER" "4"

  Unknown arguments specified
```

I get this error, not sure why. 

linux x86-64.

**Comment by calvin2021y** [comment]
only get this error with lto build with `set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")`.

without CMAKE_TRY_COMPILE_TARGET_TYPE, I get other problem.

**Comment by calvin2021y** [comment]
I add  `set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")` to fix `CMake Error: try_run() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_WRITABLE_ARGV_EXITCODE (advanced)` for other platform.  

seems not work

**Reply by jay** [reply]
I think that was fixed in #11130
